dvc/utils/fs.py

```diff
@@ -73,9 +77,9 @@ def __init__(self, path, base_path):
         super().__init__(msg)
 
 
-def contains_symlink_up_to(path, base_path):
-    base_path = os.fspath(base_path)
-    path = os.fspath(path)
+def contains_symlink_up_to(path: "StrPath", base_path: "StrPath"):
+    base_path = os.path.normcase(os.fspath(base_path))
+    path = os.path.normcase(os.fspath(path))
 
     if base_path not in path:
```

Comment:
note that we don't use `path_isin` here since it normalizes paths (not only case) and would resolve symlinks